File: C/problem003.c
/*
 * Problem 3 - Project Euler
 * http://projecteuler.net/index.php?section=problems&id=003
 */
#include <math.h>
#include <stdio.h>

#define TARGET_VALUE 600851475143LL

void problem003() {
    int ans;
    int divider;
    long long int target = TARGET_VALUE;
    double loop_max;

    /* solve this problem here */
    divider = 2;
    loop_max = floor(sqrt(target));
    while (divider <= loop_max) {
        if (target % divider == 0) {
            target /= divider;
            loop_max = floor(sqrt(target));
        } else {
            divider++;
        }
    }

    ans = (int)target;

    printf("%s: answer = %d\n", __FUNCTION__, ans);
}

int main(int argc, char* argv[]) {
    problem003();
    return 0;
}

File: Zig/problem003.zig
// zig version 0.13.0
const std = @import("std");
const stdout = std.io.getStdOut().writer();

// #3 Largest prime factor - Project Euler
// http://projecteuler.net/index.php?section=problems&id=3
pub fn main() !void {
    var target: u64 = 600851475143;
    var ans: u64 = 0;
    var i: u64 = 2;

    while (i * i <= target) : (i += 1) {
        while (target % i == 0) {
            ans = i;
            target /= i;
        }
    }

    if (target > ans) {
        ans = target;
    }

    try stdout.print("{}\n", .{ans});
}
